packages/zql/src/query/query-registry.ts

```diff
@@ -327,26 +330,29 @@ function createCustomQueryBuilder<
   T extends keyof S['tables'] & string,
   R,
   C,
-  Args extends ReadonlyJSONValue | undefined,
+  ArgsInput extends ReadonlyJSONValue | undefined,
+  ArgsOutput extends ReadonlyJSONValue | undefined,
   HasArgs extends boolean,
 >(
-  queryDef: QueryDefinition<S, T, R, C, any, Args>,
+  queryDef: QueryDefinition<S, T, R, C, ArgsInput, ArgsOutput>,
   name: string,
-  args: Args,
+  inputArgs: ArgsInput,
+  validatedArgs: ArgsOutput,
```

Comment:
You do not really need to pass down both input and output args. It is cleaner and easier to understand to pass both though.